client/client.go

```diff
@@ -201,14 +202,20 @@ func (client *Client) WarningsSummary() (count int, timestamp time.Time) {
 	return client.warningCount, client.warningTimestamp
 }
 
+// RequestError is returned when there's an error creating an http.Request.
 type RequestError struct{ error }
 
+// Error implements the error interface.
 func (e RequestError) Error() string {
 	return fmt.Sprintf("cannot build request: %v", e.error)
 }
 
-type ConnectionError struct{ error }
+// ConnectionError represents a connection or communication error.
+type ConnectionError struct {
+	error
+}
 
+// Error implements the error interface.
```

Comment:
Thanks for the review! I know what you're getting at, and in regular code I can't stand `i++ // increment i` type of comments.

However, these feed directly into the online pkg.go.dev docs, and I think it's important to be complete there, even if sometimes redundant and obvious. One person's obvious will be another person's "why isn't this documented? does this do what I think it does?".

Another thing my suggested approach has going for it: no more debating which methods should be commented and which not. The only exceptions I think we should make are for `String()` and `Error()`, as they're builtin/stdlib interfaces common to all Go projects.

Thoughts?

Reply:
It sounds like we are in complete agreement, so it's not clear what the push-back or the "However" here is.  The rule I follow and encourage is simple: don't write comments which are obvious. This is important because not only it prevents us writing stuff that we and others will need to read without adding anything to their understanding, but it actually encourages writing good comments: if the comment for a function or type is obvious, that's often a hint that we're not doing a good job explaining what's not obvious. If there's really nothing to explain, then the comment is irrelevant.

These points apply equally to both the code and the doc site.